Refactor GeometryAccessor trait signature to return reference

zhangfengcdt opened the pull request.

zhangfengcdt committed:
Recreate GeometryAccessor trait signature to return reference

Kontinuation reviewed (approved):
This looks good for our own usage in sedona-db. The only concern is that it limits the flexibility for other use cases.

zhangfengcdt committed:
use associated type approach for GeometryAccessor

AsGeometry trait:

- Provides a way to convert both owned and borrowed geometries to &Geometry<f64>
- Implemented for both Geometry<f64> and &Geometry<f64>
- Enables zero-copy access regardless of storage strategy

zhangfengcdt committed:
fix fmt issue

zhangfengcdt committed:
Revert "fix fmt issue"

This reverts commit 5cd054670b411de23eff96870fb2c9d8c4bf3a79.

zhangfengcdt committed:
Revert "use associated type approach for GeometryAccessor"

This reverts commit 97d29a3d37c401bfaddc44e81b353d8a03096c89.
